omf-python/src/file/reader.rs

```diff
@@ -261,53 +460,85 @@ impl PyReader {
     }
 
     /// Read a Boolean array.
-    pub fn array_booleans(&self, array: &PyBooleanArray) -> PyResult<Vec<Option<bool>>> {
-        self.0
-            .array_booleans(&array.0)
-            .map_err(OmfException::py_err)?
-            .collect::<Result<Vec<_>, _>>()
-            .map_err(OmfException::py_err)
+    pub fn array_booleans<'py>(
+        &self,
+        py: Python<'py>,
+        array: &PyBooleanArray,
+    ) -> PyResult<(Bound<'py, PyAny>, Bound<'py, PyAny>)> {
+        nullable_pyarray1_from_iter(
+            py,
+            self.0
+                .array_booleans(&array.0)
+                .map_err(OmfException::py_err)?,
+        )
+        .map(|(a, b)| (a.into_any(), b.into_any()))
     }
 
     /// Read a Boundary array.
-    pub fn array_boundaries(&self, array: &PyBoundaryArray) -> PyResult<Vec<f64>> {
-        let numbers_f64 = self
+    pub fn array_boundaries<'py>(
+        &self,
+        py: Python<'py>,
+        array: &PyBoundaryArray,
+    ) -> PyResult<Bound<'py, PyAny>> {
+        // TODO: using value() to get the boundary value is lossy.
```

Comment:
Any thoughts on how we ought to transform arrays of Boundary values to numpy arrays? The current way loses the distinction between than `Less` and `LessEqual` values, which is presumably important.

Reply:
It should probably be returning a list of tuples or objects rather than an array. It would be surprising for there to be more than a hundred of them, so performance will be fine.

Reply:
We’re now returning a list of `(value, BoundaryType)` tuples.